components/Dropzone/Dropzone.tsx

```diff
@@ -0,0 +1,34 @@
+import { Group, Text, useMantineTheme, MantineTheme, Center } from '@mantine/core';
+import { Upload, Photo, X, Icon as TablerIcon } from 'tabler-icons-react';
+import { Dropzone, DropzoneStatus, IMAGE_MIME_TYPE } from '@mantine/dropzone';
+
+function getIconColor(status: DropzoneStatus, theme: MantineTheme) {
+  return status.accepted
+    ? theme.colors[theme.primaryColor][theme.colorScheme === 'dark' ? 4 : 6]
+    : status.rejected
+    ? theme.colors.red[theme.colorScheme === 'dark' ? 4 : 6]
+    : theme.colorScheme === 'dark'
+    ? theme.colors.dark[0]
+    : theme.colors.gray[7];
+}
+
+function ImageUploadIcon({
+  status,
+  ...props
+}: React.ComponentProps<TablerIcon> & { status: DropzoneStatus }) {
+  if (status.accepted) {
+    return <Upload {...props} />;
+  }
+
+  if (status.rejected) {
+    return <X {...props} />;
+  }
+
+  return <Photo {...props} />;
+}
```

Comment:
Please rename file to `ImageUploadIcon` and move the `dropzoneChildren` logic to a dedicated file, in order to use a dedicated file for each React component.